behave/contrib/scenario_autoretry.py

```diff
@@ -48,17 +49,28 @@ def patch_scenario_with_autoretry(scenario, max_attempts=3):
 
     :param scenario:        Scenario or ScenarioOutline to patch.
     :param max_attempts:    How many times the scenario can be run.
+    :param delay:           Time to wait between scenario retries.
+    :param backoff:         Backoff factor to increase the delay per retry.
     """
     def scenario_run_with_retries(scenario_run, *args, **kwargs):
+        _delay = delay
+        delay_message = ""
         for attempt in range(1, max_attempts+1):
             if not scenario_run(*args, **kwargs):
                 if attempt > 1:
                     message = u"AUTO-RETRY SCENARIO PASSED (after {0} attempts)"
                     print(message.format(attempt))
                 return False    # -- NOT-FAILED = PASSED
+
             # -- SCENARIO FAILED:
             if attempt < max_attempts:
-                print(u"AUTO-RETRY SCENARIO (attempt {0})".format(attempt))
+                if delay:
+                    if attempt > 1: _delay *= backoff
```

Comment:
Please move the assignment to the next line. Single-line `if`s are known as bad style.